### DISCUSSION TITLE
YouTube URL opens video in FreeTube app

### DISCUSSION THREAD

**Discussion by chriscjcj** [johnste/finicky#327] [General]
I'm close but not 100% there. When I click on a youtube.com or youtu.be link, I would like it to open in the [FreeTube](https://freetubeapp.io/) application. When I use the config below I can get FreeTube to launch when I go to a qualifying URL, but the video won't load. 

    {
      match: "https://youtu.be/*",
      url: { protocol: "youtube" },
      browser: "FreeTube",
    }

Many thanks for any suggestions.

**Comment by chriscjcj** [comment]
SOrry to bother everyone..... Fixed it. 

    {
      match: "https://youtu.be/*",
      url: { protocol: "https" },
      browser: "FreeTube",
    }

Don't know what I was thinking. 🙄 This seems to work.

**Reply by chriscjcj** [reply]
I worked with chat GPT to help me build a regular expression to cover all of YouTube's urls.

    {
      match: /^https?:\/\/(?:[^\/\n\s]+\.)?(youtube\.com|youtu\.be)/,
      url: { protocol: "https" },
      browser: "FreeTube",
    }

This works great for clicking links from outside of a browser. What I can't figure out is how to make FreeTube load the YouTube url if I'm clicking the link from within a browser.

**Comment by VanVorobyov** [comment]
Hi, do you have any solutions to clicking on a link within a browser? I have the same problem.
